src/app/static/src/app/components/plots/FilterSelect.vue

```diff
@@ -7,17 +7,22 @@
                     :value=treeselectValue
                     :disabled=disabled
                     :placeholder=placeholder
-                    @input="select"></treeselect>
+                    @input="input"
+                    @select="select"
+                    @deselect="deselect"></treeselect>
     </div>
 </template>
 
 <script lang="ts">
     import Vue from "vue";
     import Treeselect from '@riophae/vue-treeselect';
-    import {NestedFilterOption} from "../../generated";
+    import {FilterOption, NestedFilterOption} from "../../generated";
+    import {flattenOptions} from "../../utils";
 
     interface Methods {
-        select: (value: string[]) => void
+        input: (value: string[]) => void
```

Comment:
We've ended up with two FilterSelect components, one living with the Barchart and one at the plots level. I've modified the latter (this one) so that it can deal with both types of event emitting (string ids or full selected options - Choro uses former, Barchart and Bubble Plot the latter). I've left the Barchart FilterSelect where it is as it has extra functionality around isXAxis/isDisagg etc. Choro and Bubble Plot both using this more generic one.